### PR TITLE
Nr 341878 use reusable workflow for docker image security checks

### PR DESCRIPTION
**Pull request by rahulreddy15**
Successfully tested with the reusable workflow reusable_nightly.yaml.
PR needs to be merged after this https://github.com/newrelic/coreint-automation/pull/97 is merged.

### CODE REVIEW

File: .github/workflows/nightly.yaml [hunk 1]
```diff
@@ -1,45 +1,34 @@
 name: Nightly build
-
 on:
   schedule:
     - cron: "0 3 * * *"
   push:
     branches:
       - main
+  pull_request:
```
Comment (sigilioso) [inline .github/workflows/nightly.yaml:8]: Are we supposed to create a `nightly` image on pull-requets or is it a leftover?

Comment (rahulreddy15) [reply .github/workflows/nightly.yaml:8]: My bad. That was left over from testing...

File: .github/workflows/nightly.yaml [hunk 2]
```diff
@@ -8,38 +7,27 @@ on:
...
+    secrets:
+      docker_username: ${{ secrets.FSI_DOCKERHUB_USERNAME }}
+      docker_password: ${{ secrets.FSI_DOCKERHUB_TOKEN }}
+    with:
+      docker_image: newrelic/nri-prometheus
+      docker_tag: nightly
+      target_branches: "main"
+      integration_name: "prometheus"
+      build_command: |
+        make release
```
Comment (rajrohanyadav) [inline .github/workflows/nightly.yaml:27]: Why do we need it to be multiline when we just have a single line of input?

File: .github/workflows/nightly.yaml [hunk 3]
```diff
@@ -8,38 +7,27 @@ on:
...
+      setup_qemu: true
+      setup_buildx: true
+      setup_go: true
```
[the remark this answers is not in the document]
Comment (rahulreddy15) [reply .github/workflows/nightly.yaml:30]: Some don't need buildx and build  with a command. Buildx does more than building. Same with others.
There are exceptions so I had to include them.

File: .github/workflows/nightly.yaml (hunk 2 above)
Comment (rahulreddy15) [reply .github/workflows/nightly.yaml:27]: That was left over from copying configs from repos where multiline command were used.

**Comment by rahulreddy15** [comment]
> I'm wondering why this change is required? Will we use this workflow in some other repos?

Yeah. All the integrations that release docker images will use it.

File: .github/workflows/nightly.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (rajrohanyadav) [reply .github/workflows/nightly.yaml:30]: > Some don't need buildx and build with a command

AFAIK, `buildx` is needed when we need to build multi-arch images. And we mostly do that. I'm not sure when we wouldn't need it. But not a major thing.

File: .github/workflows/nightly.yaml (hunk 2 above)
Comment (rajrohanyadav) [reply .github/workflows/nightly.yaml:27]: Can we update it?

**Review by rajrohanyadav — APPROVED** [review]
LGTM

**Pull request merged**